ios/Example/Tests/TrifleTests.swift

```diff
@@ -146,17 +147,24 @@ WLq6hmtSmQ==
         let deviceCertificate = Certificate(version: 0, certificate: deviceCertEncoded2)
         let rootCertificate = Certificate(version: 0, certificate: rootCertEncoded)
 
-        // cert chain of length 2
-        let signDataWithRoot = try trifle.createSignedData(data: data,
-                                    keyHandle: keyHandle,
-                                    certificates: [deviceCertificate, rootCertificate])
-        XCTAssertNotNil(signDataWithRoot, "This is TODO - once this is validated, this test should FAIL")
         
         // cert chain of length 1
         let signDataWithoutRoot = try trifle.createSignedData(data: data,
                                      keyHandle: keyHandle,
                                      certificates: [deviceCertificate] )
         XCTAssertNotNil(signDataWithoutRoot, "This is TODO - once this is validated, this test should FAIL")
+
+        
+        let envData = try ProtoDecoder().decode(SignedData.EnvelopedData.self, from: signDataWithoutRoot.enveloped_data!)
+        XCTAssertEqual(envData.version, 0)
+        XCTAssertEqual(envData.data, data)
+        XCTAssertEqual(envData.signing_algorithm, SignedData.Algorithm.ECDSA_SHA256)
+
+        // cert chain of length 2
+        let signDataWithRoot = try trifle.createSignedData(data: data,
+                                    keyHandle: keyHandle,
+                                    certificates: [deviceCertificate, rootCertificate])
```

Comment:
copy/paste error?  why should this one fail?

Reply:
It should fail because we are not verifying that the public key associated with the key handle is the same one in the deviceCertificate. Once we add this validation, we have to make sure that we have a proper end 2 end test (ie the cert is bound to the key pair that is generated in that test).